File: First_Index_Occurence/First_index_occurence.py
def first_index_occurence(haystack, needle):

    first_index_word = []
    first_index = ""
    if needle in haystack:
        for i in range(len(haystack)):
            if needle[0] == haystack[i]:
                first_index_word.append(i)
                first_index = first_index_word[0]
        print(f"First index occurence of the needle word = {first_index}")
    else:
        print(-1)



first_index_occurence("sadbutsad", "sad")
first_index_occurence("leetcode", "leeto")

File: First_Index_Occurence/firstIndexOccurence.java
import java.util.ArrayList;
import java.util.List;

public class firstIndexOccurence{
    public static void main(String[] args) {
        process("sadbutsad", "sad");
        process("leetcode", "leeto");
        process("helloworld", "world");
    }
    private static void process(String haystack, String needle) {
        List<Integer> firstIndexWord = new ArrayList<>();
        int firstIndex = -1;
        if(haystack.contains(needle)) {
            for (int i = 0; i <= haystack.length() - needle.length(); i++) {
                // Check if the substring from index 'i' matches the needle
                if (haystack.substring(i, i + needle.length()).equals(needle)) {
                    firstIndexWord.add(i);  // Add the index to the list
                    firstIndex = firstIndexWord.get(0);  // Get the first index
                    break;  // Stop after finding the first occurrence
                }
            }
            System.out.println("First index occurrence of the needle word = " + firstIndex);
        }
        else {
            System.out.println(-1);
        }

    }
}
